components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/FunctionCmd.java

```diff
@@ -0,0 +1,100 @@
+package org.wso2.apimgt.gateway.cli.cmd;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.apimgt.gateway.cli.exception.CLIRuntimeException;
+import org.wso2.apimgt.gateway.cli.utils.GatewayCmdUtils;
+import org.wso2.apimgt.gateway.cli.utils.RouteUtils;
+
+import java.io.File;
+import java.io.PrintStream;
+import java.util.List;
+
+@Parameters(commandNames = "function", commandDescription = "add function")
+public class FunctionCmd implements GatewayLauncherCmd {
+
+    private static final Logger logger = LoggerFactory.getLogger(FunctionCmd.class);
+    private static PrintStream outStream = System.out;
+
+    @Parameter(hidden = true, required = true)
+    private List<String> mainArgs;
+
+    @Parameter(names = {"-add", "--add function"}, hidden = true)
+    private Boolean addFunction = false;
+
+    @Parameter(names = {"-update", "--update function"}, hidden = true)
+    private Boolean updateFunction = false;
+
+    @Parameter(names = {"-l", "--fileName"}, hidden = true)
+    private String fileLocation;
+
+    @Parameter(names = {"-ai", "--api id"}, hidden = true)
+    private String apiID;
+
+    @Parameter(names = {"-ri", "--resource id"}, hidden = true)
+    private String resourceID;
+
+    @Parameter(names = {"-in", "--function in"}, hidden = true)
+    private String inFunction;
+
+    @Parameter(names = {"-out", "--function out"}, hidden = true)
+    private String outFunction;
+
+    @Override
+    public void execute() {
+
+        String projectName = GatewayCmdUtils.getProjectName(mainArgs);
+        File projectLocation = new File(GatewayCmdUtils.getProjectDirectoryPath(projectName) + "/src");
+
+        if (!projectLocation.exists()) {
+            throw new CLIRuntimeException("Project " + projectName + " does not exist.");
+        }
+        //copy .bal file to the project directory
+        try {
+            File source = new File(fileLocation);
+            FileUtils.copyFileToDirectory(source, projectLocation);
+            outStream.println("file copied");
+        } catch (Exception e) {
+            logger.error("error occured while copying file:" + e);
+        }
+
+        if (addFunction || updateFunction) {
```

Comment:
I only see add and update function actions? Is it neccessory to check it, if it can only be one of these two?
If theres a need to check it, return in `false` case and continue the logic in `true` case.
```suggestion
        if (!(addFunction || updateFunction)) {
            return;
        }
```

Reply:
This is to be extended with delete function as well. Updated with the requested change.